src/queue.jl

```diff
@@ -121,31 +121,42 @@ function get_default_queue(device::ROCDevice)
     end
 end
 
-"Kills all kernels executing on the given queue, and destroys the queue."
+"""
+    kill_queue!(queue::ROCQueue; force=false)
+
+Kills all kernels executing on the given queue, destroys the queue,
+and performs internal clean up. Returns true upon success.
+
+If `force` is `true`, this will attempt to kill a queue even if a previous
+attempt has already been made. Since queue pointers may be reused by the device,
+this may cause unpredictable behaviour by destroying an active queue.
+"""
 function kill_queue!(queue::ROCQueue; force=false)
-    #queue_ptr = @atomicswap queue.queue = Ptr{HSA.Queue}(0)
-    queue_ptr = queue.queue
```

Comment:
I don't fully understand the workflow you have in mind here: I removed it because I assumed the queue pointer wouldn't change.

Isn't allowing this pointer to change elsewhere asking for trouble? If it's set to NULL at some point during `kill_queue()`, does that mean our cached reference to it is still okay? Could we now inadvertently be killing a new queue with a reused pointer? Seems like we're making life hard from the perspective of thread safety. (For that matter, I also feel like the 'force' option here is asking for trouble).